test/integrationtests/voight_kampff/test_setup.py

```diff
@@ -205,7 +205,11 @@ def main(cmdline_args):
     This installs and/or upgrades any skills needed for the tests and
     collects the feature and step files for the skills.
     """
-    args = get_arguments(cmdline_args)
+    is_pre_parsed = type(cmdline_args).__name__ == 'Namespace'
+    if is_pre_parsed:
+        args = cmdline_args
+    else:
+        args = get_arguments(cmdline_args)
```

Comment:
The type check looks abit weird, maybe this block could be changed to
```python
if not isinstance(args, argparse.Namespace):
    args = get_arguments(cmdline_args)
else:
    args = cmdline_args

alternatively the get_arguments(cmdline_args) can be moved to be an argument for main()

```python
if __name__ == '__main__':
    main(get_arguments(cmdline_args))
```

Reply:
Yeah nice, I've shifted the argument parsing to the call of main as this seems like the cleanest approach.

Thanks